packages/makeup-menu-button/src/index.js

```diff
@@ -117,14 +121,22 @@ function _onMenuKeyDown(e) {
 }
 
 function _onMenuItemSelect(e) {
-  if (this._buttonPrefix && e.detail.el.getAttribute("role") === "menuitemradio") {
-    this._buttonTextEl.innerText = `${this._buttonPrefix} ${e.detail.el.innerText}`;
-  }
-
   const widget = this;
 
   setTimeout(function () {
     widget._expander.expanded = false;
     widget._buttonEl.focus();
   }, 150);
+
+  if (e.detail.el.getAttribute("role") !== "menuitemradio") {
+    return;
+  }
+
+  const valueSelector = e.detail.el.querySelector(this._options.valueSelector);
+  if (valueSelector) {
+    const valueType = this._options.valueTypeHTML ? "innerHTML" : "innerText";
+    this._buttonTextEl[valueType] = valueSelector[valueType];
```

Comment:
This disregards any prefix that might exist, is it an anti-pattern to use both a prefix and `valueTypeHTML`?

Reply:
Yeah, I assumed when you have a selector, you have more ability to include all the elements in it. Adding a prefix is unnecessary but I don't have a strong preference for this.

Reply:
Ok, if you want to keep it this way then it should be documented somewhere that you can't use both prefix and valueTypeHTML

Reply:
Refactored code to consider both prefixed text and Icon.